src/c-writer.cc

```diff
@@ -1292,7 +1292,14 @@ void CWriter::Write(const Const& const_) {
         // Negative zero. Special-cased so it isn't written as -0 below.
         Writef("-0.0");
       } else {
-        Writef("%.17g", Bitcast<double>(f64_bits));
+        char buf[128];
+        snprintf(buf, sizeof(buf), "%.17g", Bitcast<double>(f64_bits));
+        // Append .0 to if sprint didn't include a decimal point.
+        // Works around https://github.com/WebAssembly/wabt/issues/2422
+        if (!strchr(buf, '.') && !strchr(buf, '+')) {
```

Comment:
`'+'`?

Reply:
This avoids adding `.0` to constants that look like `1.2200011354000000e+28`

Reply:
but that has a `.` already, do you mean like `1e-100`?

Reply:
Updated to look for `e` instead, and added a comment.